src/EnergyPlus/SwimmingPool.cc

```diff
@@ -460,6 +420,61 @@ namespace SwimmingPool {
+            ShowContinueError(cAlphaField2 + "=\"" + Alpha2 + "\" has been used in another radiant system, ventilated slab, or pool.");
+            ShowContinueError(
+                "A single surface can only be a radiant system, a ventilated slab, or a pool.  It CANNOT be more than one of these.");
+            ErrorsFound = true;
+            // Something present that is not allowed for a swimming pool (non-CTF algorithm, movable insulation, or radiant source/sink
+        } else if (DataSurfaces::Surface(Pool(Item).SurfacePtr).HeatTransferAlgorithm != DataSurfaces::HeatTransferModel_CTF) {
+            ShowSevereError(DataSurfaces::Surface(Pool(Item).SurfacePtr).Name +
+                            " is a pool and is attempting to use a non-CTF solution algorithm.  This is "
+                            "not allowed.  Use the CTF solution algorithm for this surface.");
+            ErrorsFound = true;
+        } else if (DataSurfaces::Surface(Pool(Item).SurfacePtr).Class == DataSurfaces::SurfaceClass_Window) {
+            ShowSevereError(DataSurfaces::Surface(Pool(Item).SurfacePtr).Name +
+                            " is a pool and is defined as a window.  This is not allowed.  A pool must be a floor that is NOT a window.");
+            ErrorsFound = true;
+        } else if (DataSurfaces::Surface(Pool(Item).SurfacePtr).MaterialMovInsulInt > 0) {
+            ShowSevereError(DataSurfaces::Surface(Pool(Item).SurfacePtr).Name +
+                            " is a pool and has movable insulation.  This is not allowed.  Remove the movable insulation for this surface.");
+            ErrorsFound = true;
+        } else if (DataHeatBalance::Construct(DataSurfaces::Surface(Pool(Item).SurfacePtr).Construction).SourceSinkPresent) {
+            ShowSevereError(
+                DataSurfaces::Surface(Pool(Item).SurfacePtr).Name +
+                " is a pool and uses a construction with a source/sink.  This is not allowed.  Use a standard construction for this surface.");
+            ErrorsFound = true;
+        } else { // ( Pool( Item ).SurfacePtr > 0 )
+            DataSurfaces::Surface(Pool(Item).SurfacePtr).PartOfVentSlabOrRadiantSurface = true;
+            DataSurfaces::Surface(Pool(Item).SurfacePtr).IsPool = true;
+            Pool(Item).SurfaceToPoolIndex(Pool(Item).SurfacePtr) = Item;
+            Pool(Item).ZonePtr = DataSurfaces::Surface(Pool(Item).SurfacePtr).Zone;
+            // Check to make sure pool surface is a floor
+            if (DataSurfaces::Surface(Pool(Item).SurfacePtr).Class != DataSurfaces::SurfaceClass_Floor) {
+                ShowSevereError(RoutineName + CurrentModuleObject + "=\"" + Alpha1 + " contains a surface name that is NOT a floor.");
+                ShowContinueError(
+                    "A swimming pool must be associated with a surface that is a FLOOR.  Association with other surface types is not permitted.");
+                ErrorsFound = true;
+            }
+        }
+    }
```

Comment:
Nice, this is a very complete unit test. Where do you check if the surface is referenced by another pool? I didn't look but is that done after all pools are read in?

Reply:
@rraustad It's "Test 2" where PartOfVentSlabOrRadiantSurface is set to true.  That also tracks pool as well as ventilated slabs and radiant systems.  That variable is set to true in the new subroutine when all the checks pass in the "else" of the if block.

Reply:
So a pool is a ventilated slab? or is it a radiant surface? Yes, I do see in the new function where this var is set true for pools.

    } else { // ( Pool( Item ).SurfacePtr > 0 )
       DataSurfaces::Surface(Pool(Item).SurfacePtr).PartOfVentSlabOrRadiantSurface = true;

I see 14 matches. What do you think of PartOf(or Is)VentSlabOrRadSurfOrPool ?? Won't hold you to this, but I was confused, so will another developer in the future also be confused?

I'm not sure this answers my question. If I have an input file with 2 pools and I specify the same surface name for both, is there an error?

Reply:
I agree, this is a great example of extracting out a reasonable amount of code into a standalone function that is easily tested.  One step that could've taken this over the top would've been to make this function a method on the pool struct so that inside the function you never had to look up `Pool(Item)` like you do a bunch here.  If you do end up needing to do anything else to resolve @rraustad 's comments here, I would suggest you take a couple minutes and make that change.  If no more commits are needed, don't sweat it.

Reply:
@rraustad @Myoldmopar Ok, I've committed changes to address this request by @Myoldmopar and brought the code up to speed with develop.  I think this is now good to go though I understand that it might need another round of reviews given the number of changes that I made.  The changes should be algorithm neutral (except for the elimination of an unused variable).  Let me know if there's anything else that needs to be done here.  I am hoping that CI will come back clean and this is ready to go.